pinot-core/src/main/java/org/apache/pinot/core/operator/filter/RangeIndexBasedFilterOperator.java

```diff
@@ -44,44 +53,74 @@ public class RangeIndexBasedFilterOperator extends BaseFilterOperator {
 
   private static final String EXPLAIN_NAME = "FILTER_RANGE_INDEX";
 
-  private final RangeEvaluator _rangeEvaluator;
-  private final PredicateEvaluator _rangePredicateEvaluator;
+  static final Set<FieldSpec.DataType> SUPPORTED_RAW_DATA_TYPES =
+      EnumSet.of(FieldSpec.DataType.INT, FieldSpec.DataType.LONG, FieldSpec.DataType.FLOAT, FieldSpec.DataType.DOUBLE);
+
+  private final RangeIndexReader<ImmutableRoaringBitmap> _rangeIndexReader;
+  private final PredicateEvaluator _predicateEvaluator;
   private final DataSource _dataSource;
+  private final FieldSpec.DataType _parameterType;
+  private final Object _parameters;
   private final int _numDocs;
 
+  static boolean canEvaluate(PredicateEvaluator predicateEvaluator, DataSource dataSource) {
+    if (dataSource.getRangeIndex() != null) {
+      boolean datatypeSupported = (predicateEvaluator.isDictionaryBased() || SUPPORTED_RAW_DATA_TYPES.contains(
```

Comment:
After a second thought, I feel we can remove most of the checks here. When range index exists, it implies that the column is either dictionary encoded, or has supported data types. All we need to check is for EQ the range index should be exact.
Although this introduce some assumptions to this module, it reduces the per-segment check overhead at query time.

Reply:
Self documenting code tends to avoid implicit dependencies like that. This way, you have a _declaration of requirements_ in one place which reduces cognitive overhead for the reader. The overhead of these kinds of checks is also rather small, especially if the method gets inlined and some of the redundancy at the call site gets removed.

Reply:
Against my better judgement, I have removed the majority of the checks here.

Reply:
I would vote against removing these checks. They shouldn't be that heavy and as appointed by @richardstartin, they made the code easier to understand and even more resilient without an expected impact on performance.